Add LaTeX CV template and fonts, migrate CV data to YAML format

frank-alvarado opened the pull request.

frank-alvarado committed:
Add LaTeX CV template and fonts, migrate CV data to YAML format

frank-alvarado committed:
Refactor CV build system with containerized LaTeX and simplified scripts

frank-alvarado committed:
Add Docker CI/CD for building CV image and generating PDF

frank-alvarado committed:
Remove CV PDF generation and artifact upload steps from Docker CI workflow

frank-alvarado committed:
Add Jest JUnit reporter and simplify Jest config

frank-alvarado committed:
Add workflow_dispatch trigger to enable manual workflow runs in GitHub Actions

frank-alvarado committed:
Merge remote-tracking branch 'origin/main' into latex-cv-docker

frank-alvarado committed:
Replace Makefile build with direct XeLaTeX commands for CV generation

frank-alvarado committed:
chore: update package-lock.json dependencies

fern-alva commented:
looks good

fern-alva commented on latex/Dockerfile:
```diff
@@ -0,0 +1,23 @@
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install TeX Live (XeLaTeX), Python, pip, Git, and fonts
+RUN apt-get update && apt-get install -y \
```
why did you do this?

frank-alvarado committed:
Enhance documentation and CI/CD for CV generation

- Added an Architecture Overview section to README.md to illustrate the CV generation process.
- Introduced requirements.txt for managing Python dependencies, specifically PyYAML.
- Updated SECURITY.md to clarify vulnerability reporting and supported versions.
- Implemented a new GitHub Actions job to generate the CV PDF as part of the CI/CD pipeline.
- Refactored Terraform configuration to use a module for static site deployment, improving organization and maintainability.
- Added tests for the ThemeToggle component to ensure proper functionality and persistence.
- Updated Next.js configuration to enhance security headers and removed unnecessary page extensions.
